tests/flexmock_test.py

```diff
@@ -49,15 +53,36 @@ def instance_method(self, a):
         pass
 
 
-def assert_raises(exception, method, *kargs, **kwargs):
+@contextmanager
+def assert_raises(expected_exception: Type[BaseException], match: Union[RE_TYPE, str]):
+    """Assert that code raises the correct exception with a correct error message.
+
+    Args:
+        expected_exception: Type of the expected exception.
+        match: String or pattern to match the error message against.
+    """
     try:
-        method(*kargs, **kwargs)
-    except exception:
-        return
-    except Exception:
-        instance = sys.exc_info()[1]
-        print("%s" % instance)
-    raise Exception("%s not raised" % exception.__name__)
+        yield
+    except Exception as raised_exception:
+        if not isinstance(raised_exception, expected_exception):
+            raise AssertionError(
+                f"Expected exception '{type(expected_exception)}' "
+                f"but '{type(raised_exception)}' was raised"
+            ) from raised_exception
+        if match is not None:
+            fail = False
+            if isinstance(match, RE_TYPE):
+                fail = not match.search(str(raised_exception))
+                match = match.pattern
+            else:
+                fail = str(raised_exception) != str(match)
```

Comment:
Should this case be a non-compiled regex ?

Reply:
What do you mean?

Reply:
I mean that if you put a text there, we don't treat it as a text that should be compared verbatim but as a regex, so it would be possible to put wildcards and whatnot

Reply:
```suggestion
                fail = not re.fullmatch(str(match), str(raised_exception)) 
```
(not sure we need to use `str` on the match since it's *supposed* to be a string but eh)

Reply:
If we would treat any string as a regex, that could cause issues when characters would be unintentionally interpreted as regex metacharacters if they are not escaped.

Reply:
Not sure if this is what you were thinking but I created a related issue to make a change to the public API: https://github.com/flexmock/flexmock/issues/78